openhcl/shared_pool_alloc/src/lib.rs

```diff
@@ -266,6 +267,11 @@ impl user_driver::vfio::VfioDmaBuffer for SharedPoolAllocator {
             .map_file(0, len, gpa_fd.get(), file_offset, true)
             .context("unable to map allocation")?;
 
+        // SAFETY: The previous call to memmap should have made exactly len bytes at offset 0 valid to read and write
+        unsafe {
```

Comment:
Could this be made a method on the mapping? Like `fn zero(&mut self)` or something similar? Then we wouldn't need the allow(unsafe) in this file, right?

Reply:
I think we want the guarantee that VfioDmaBuffers are always zeroed on construction. But yes I think we should be pushing the unsafe code further down here.

Reply:
I think it's OK to use .zero() to avoid this unsafe. We will have cases soon where we will be restoring on top of existing data, so there can be no requirement that says you can't construct a MemoryBlock that has non-zero data.